### PR TITLE
Show toast when no updates are available.

### PR DESCRIPTION
**Pull request by Isira-Seneviratne**
#### What is it?
- [ ] Bugfix (user facing)
- [x] Feature (user facing)
- [ ] Codebase improvement (dev facing)
- [ ] Meta improvement to the project (dev facing)

#### Description of the changes in your PR

- Display a toast if no new versions of NewPipe are available when checking for updates.

#### Before/After Screenshots/Screen Record

- Before:
- After:

#### Fixes the following issue(s)

- Fixes #

#### APK testing 

The APK can be found by going to the "Checks" tab below the title. On the left pane, click on "CI", scroll down to "artifacts" and click "app" to download the zip file which contains the debug APK of this PR.

#### Due diligence
- [x] I read the [contribution guidelines](https://github.com/TeamNewPipe/NewPipe/blob/HEAD/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by TacoTheDank** [comment]
Wouldn't this show every time the app is launched? That would be pretty annoying, I think.

**Comment by mhmdanas** [comment]
I agree with @TacoTheDank, it makes little sense to me to show a notification if there is _no_ update available, plus it would be quite irritating. What problem are you trying to solve by this PR @Isira-Seneviratne?

**Comment by Isira-Seneviratne** [comment]
> I agree with @TacoTheDank, it makes little sense to me to show a notification if there is _no_ update available, plus it would be quite irritating. What problem are you trying to solve by this PR @Isira-Seneviratne?

The idea was to show some feedback if no update was available, since at the moment nothing is shown if the app is up-to-date.

**Comment by mhmdanas** [comment]
I think that would be more suited for a manual update check perhaps.

**Comment by TobiGr** [comment]
I would not show a notification, but rather a toast. A toast saying "Checking for updates..." is shown when the "check for updates" button is pressed (settings > updates).
I agree with the others that this info should not be shown when the check was performed automatically.

**Comment by Isira-Seneviratne** [comment]
@TobiGr Yeah, I'll make that change.

File: app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java:35]: I think you should use the MANUAL flag inside `NewVersionWorker.checkNewVersion` to determine whether to ignore the expiry time, instead of resetting the expiry time here. Also, since only one statement would be left here then, I would remove `checkNewVersionNow()` and inline the work enqueueing.

File: app/src/main/res/values/strings.xml
Comment (Stypox) [inline app/src/main/res/values/strings.xml:556]: Since you added `%s` to this string, I think you should remove all of the copies in other languages so that they are retranslated with the `%s` in the correct position.

**Review by Stypox — CHANGES_REQUESTED** [review]
Looks almost good to me

File: app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt:168]: This is not what I meant: remove this and just add an `if (!manual)` around [these lines](https://github.com/TeamNewPipe/NewPipe/blob/6ca5b4108ba98274ab7602bbbb1ebee03439147a/app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt#L81-L87)

**Review by Stypox — CHANGES_REQUESTED** [review]

File: app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt
Comment (Isira-Seneviratne) [reply app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt]: Is the change okay?

File: app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt:49]: No lines longer than 100 characters... ktlint seems to be misconfigured, it should detect these problems
```suggestion
                Toast.makeText(applicationContext, R.string.app_update_unavailable_toast,
                        Toast.LENGTH_SHORT).show()
```

File: app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/NewVersionWorker.kt:68]:
```suggestion
            .setContentTitle(applicationContext.getString(
                    R.string.app_update_available_notification_title))
            .setContentText(applicationContext.getString(
                    R.string.app_update_available_notification_text, versionName))
```

File: app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java
Comment (Stypox) [inline app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java:19]: I would make this `false` intead of `true`, since the user did not directly ask for an update check, he just enabled them. So if the last check was done a while ago (or never), then it's ok to check again, but if it has been done minutes earlier since the user just disabled and enabled twice in a row, it should not be done. Also, it's better to consider it automatic since we don't want a toast to be shown to the user in this case, as he does not expect feedback I think.

File: app/src/main/res/values/strings.xml
Comment (Stypox) [inline app/src/main/res/values/strings.xml:554]:
```suggestion
    <string name="app_update_unavailable_toast">You are running the latest version of NewPipe</string>
```

**Review by Stypox — CHANGES_REQUESTED** [review]
I get the following error

```
java.lang.NullPointerException: Can't toast on a thread that has not called Looper.prepare()
        at ...
        at org.schabi.newpipe.NewVersionWorker.compareAppVersionAndShowNotification(NewVersionWorker.kt:49)

**Comment by Isira-Seneviratne** [comment]
> I get the following error, probably the wrong thread or context are used?
> 
> ```
> java.lang.NullPointerException: Can't toast on a thread that has not called Looper.prepare()
>         at ...
>         at org.schabi.newpipe.NewVersionWorker.compareAppVersionAndShowNotification(NewVersionWorker.kt:49)

Yeah, it seems like the fix is to use a `Handler` or `Executor` to show the toast.

By the way, how can I check for updates in the debug build?

**Comment by TacoTheDank** [comment]
> By the way, how can I check for updates in the debug build?

I would assume that debug builds are for debugging, and that update checking isn't needed for them.

**Comment by Isira-Seneviratne** [comment]
> > By the way, how can I check for updates in the debug build?
> 
> I would assume that debug builds are for debugging, and that update checking isn't needed for them.
> 

Right, my bad. I'll check with a release build.

File: app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java
Comment (TobiGr) [reply app/src/main/java/org/schabi/newpipe/settings/UpdateSettingsFragment.java:19]: I'd keep this `true` because the user enabled the check manually, in other words wants NewPipe to check for new versions. Reporting back, that we did not find any new version *this time* is ok. It shows that the app does something as requested.

**Review by TobiGr — APPROVED** [review]
LGTM. Thank you

**Comment by TobiGr** [comment]
btw. a rebase is required. I just pulled the latest translations from weblate. 

To the maintainer merging this PR: please make sure to sync weblate once this PR is merged

**Review by Stypox — APPROVED** [review]
Thanks!

**Pull request merged**
